### ISSUE TITLE
CMake generator creates an invalid file if user_info var has backslashes

### ISSUE THREAD

**Issue by mpdelbuono** [conan-io/conan#1640]
Consider a recipe that does something like:
```python
def package_info(self):
   self.user_info.var = self.package_folder
```

On Windows, the CMake generator of a consumer of that package would create a CMake variable CONAN_USER_XXXX_var that contains unescaped backslashes, which is illegal. The CMake generator should escape backslashes it encounters.

(I believe I already have a fix for this issue, but I am still verifying it.)

**Comment by mpdelbuono** [comment]
Side note: this would be a potentially breaking change if it were fixed. It is conceivable that there are recipes in the wild which work around this issue by escaping it from within conanfile.py. If so, the fix for this would break existing recipes relying on this behavior.

**Comment by memsharded** [comment]
If it is illegal, then we can assume that nobody is doing it, and users are already escaping or replacing backslashes. I cannot imagine an automatic fix for this, however the code is not that terrible:

```python
def package_info(self):
     self.user_info.var = self.package_folder.replace("\\", "/")
```

**Comment by mpdelbuono** [comment]
While the package_info() workaround above works, it imposes an inappropriate requirement upon the package: it needs to know that its consumers are going to use a CMake generator. The dependency shouldn't be aware of this information, and that's why I think the responsibility falls to the CMake generator to properly escape the values. This way all consumers will see the value as expected, and only if using CMake will the string become escaped as necessary.

(Note that upon investigation, I determined this problem actually wasn't limited to just backslashes, but also double quotes and $.)

**Comment by memsharded** [comment]
Released in 0.26
